mlflow/pyfunc/__init__.py

```diff
@@ -2098,6 +2098,20 @@ def spark_udf(
     # can't access NFS, so it needs to use `spark.addArtifact`.
     use_dbconnect_artifact = is_dbconnect_mode and not is_in_databricks_shared_cluster_runtime()
 
+    if use_dbconnect_artifact:
+        udf_sandbox_info = get_dbconnect_udf_sandbox_info(spark)
+        if Version(udf_sandbox_info.mlflow_version) < Version("2.19.0"):
+            raise MlflowException(
+                "Using 'mlflow.pyfunc.spark_udf' in Databricks Serverless or in remote "
+                "Databricks Connect requires UDF sandbox image installed with MLflow "
+                "of version >= 2.19."
+            )
+        if Version(udf_sandbox_info.runtime_version) < Version("16.0"):
```

Comment:
This like looks a little scary because run time version is not a python package version. Can we compare the versions as a tuple?

Reply:
In `get_dbconnect_udf_sandbox_info` function, it normalize `runtime_version` to format like `<major>.<minor>`, so it should be safe.

Reply:
Can you add a comment? The current code looks unsafe.

Reply:
or you can compare the major version.